pkg/translator/translator.go

```diff
@@ -73,17 +74,97 @@ func (t *Translator) MakeConfigComponent(component hpsf.Component) (config.Compo
 	// first look in the template components
 	tc, ok := t.components[component.Kind]
 	if ok {
+		// found it, manufacture a new instance of the component
 		tc.SetHPSF(component)
 		return &tc, nil
 	}
 
-	// fall back to the base components
-	switch component.Kind {
-	case "TraceGRPC", "TraceHTTP", "LogGRPC", "LogHTTP", "RefineryGRPC", "RefineryHTTP":
-		return NewInput(component)
-	default:
-		return nil, fmt.Errorf("unknown component kind: %s", component.Kind)
+	// nothing found so we're done
+	return nil, fmt.Errorf("unknown component kind: %s", component.Kind)
+}
+
+// ValidateConfig validates the configuration of the HPSF document as it stands with respect to the
+// components and templates installed in the translator.
+// Note that it returns a validation.Result so that the errors can be collected and reported in a
+// structured way. This allows for multiple validation errors to be returned at once, rather than
+// stopping at the first error. This is useful for providing feedback to users on multiple issues
+// in their configuration.
+func (t *Translator) ValidateConfig(h *hpsf.HPSF) error {
+	if h == nil {
```

Comment:
Is an empty (no components) HPSF document valid? If not, we should do a len check here too.

Reply:
I thought so, and I was wrong. At least it broke tests. So I reverted it.